lib/srv/forward/sshserver.go

```diff
@@ -1366,6 +1373,15 @@ func (s *Server) handleEnvs(ctx context.Context, ch ssh.Channel, req *ssh.Reques
 		return trace.Wrap(err, "failed to unmarshal envs")
 	}
 
+	// As a forwarder we want to capture the environment variables set by the caller.
+	// Environment variables are used to pass existing session IDs (Assist) and
+	// other flags like enabling non-interactive session recording.
+	// We want to save all environment variables even if the ssh server rejects
+	// them, as we still need their information (e.g. the session ID).
+	for k, v := range envs {
+		scx.SetEnv(k, v)
```

Comment:
Will this send the environment variables to the sshd? I think this is a big change in terms of functionality and potential performance (setting environment variables in SSH is slow).

Reply:
The environment variables are already sent to sshd, we just keep them in memory as well. I see two risks with this change:
- if a user sets a lot of XXL environment variables, this will consume more memory
- users might set those environment variables themselves, there might be a security risk if a user changes its TELEPORT_SESSION var. This PR doesn't introduce this risk: it was already here since the Assist non-interactive recording logic.